test/test_exhaust_credits

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+test_mode() {
+    executable="$1"
+    name="$2"
+    value="$3"
+    expected_ret=$4
+
+    echo "~> Testing ${name} mode"
+
+    NX_GZIP_TYPE_SELECTOR=$value $executable
+    if [ $? -ne $expected_ret ]; then
+	echo "  Failed! Expected ${expected_ret} but got $?"
+	exit 1
+    fi
+
+    # Wait some time to until all secondary process finish
+    sleep 2
+    echo
+}
+
+echo "=== Testing zlib API ==="
+# Return code 0 means no failures should occur
+test_mode ./exhaust_credits "AUTO" 0 0
+
+# Return code 1 means the number of used should not be N-1
+test_mode ./exhaust_credits "SW" 1 1
+
+# Return code 18 means deflateInit and inflateInit
```

Comment:
deflateInit and inflateInit **should fail**

Reply:
Fixed.